### ISSUE TITLE
Issues with default parameters in aws lambda node 6.10

### ISSUE THREAD

**Issue by walkerlangley** [awslabs/aws-sam-local#340]
Has anyone else had an issue with default parameters not working in functions when using node 6.10?  The following code worked just fine when running aws-sam-local, however, when deployed I got an error `"message":"Cannot match against 'undefined' or 'null'."` of which I spent the better half of the day trying to figure out what was going on since the code worked just as expected locally. Come to find out, if I didn't have the queryParam "active", active was null and the default param wasn't being reset to an empty string.  I'm wondering why this worked just fine when running aws-sam-local but it blew up once it was deployed.

What am I missing?

```javascript
const { active } = event.queryStringParameters;
const results = yield user.getDashboards(userId, active);

// in the user.js file
const getDashboards = (userId, active = '') => {
  // do some things
}
```

**Comment by michaelj-smith** [comment]
Is this a duplicate of #338 ?

When deployed to AWS, `event.queryStringParameters` is null, so trying to dereference the `active` property on that null value will result in an error.
In contrast, SAM Local always sets event.queryStringParameters to an instance of an object.

**Comment by walkerlangley** [comment]
@michaelj-smith: Yes it is.  I restructured the code to get around this, but good to know I'm not crazy.
